Error updating stats on mobile client

Stats are not updating on the mobile client. This is the application log:

01 - service created
02 - instantiate OAuth using sFactory: class org.wordpress.android.networking.OAuthAuthenticatorFactoryDefault
03 - instantiate RestClient using sFactory: class org.wordpress.android.networking.RestClientFactoryDefault
04 - update started for blogID - 63627876
05 - org.wordpress.android.ui.stats.service.StatsService$BarChartListener responded with Error
06 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
07 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
08 - org.wordpress.android.ui.stats.service.StatsService$TopPostAndPageListener responded with Error
09 - org.wordpress.android.ui.stats.service.StatsService$SearchEngineTermsListener responded with Error
10 - org.wordpress.android.ui.stats.service.StatsService$ClicksListener responded with Error
11 - org.wordpress.android.ui.stats.service.StatsService$ReferrersListener responded with Error
12 - org.wordpress.android.ui.stats.service.StatsService$ViewsByCountryListener responded with Error
13 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
14 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
15 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
16 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
17 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
18 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
19 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
20 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
21 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
22 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
23 - Stats Summary Call responded
24 - service destroyed
25 - service created
26 - update started for blogID - 63627876
27 - org.wordpress.android.ui.stats.service.StatsService$BarChartListener responded with Error
28 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
29 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
30 - org.wordpress.android.ui.stats.service.StatsService$ReferrersListener responded with Error
31 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
32 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
33 - org.wordpress.android.ui.stats.service.StatsService$TopPostAndPageListener responded with Error
34 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
35 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
36 - org.wordpress.android.ui.stats.service.StatsService$ClicksListener responded with Error
37 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
38 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
39 - org.wordpress.android.ui.stats.service.StatsService$SearchEngineTermsListener responded with Error
40 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
41 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
42 - org.wordpress.android.ui.stats.service.StatsService$ViewsByCountryListener responded with Error
43 - Error while reading Stats - java.lang.NullPointerException - exception: java.lang.NullPointerException
44 - StackTrace: com.android.volley.VolleyError: java.lang.NullPointerException<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:133)<br/>Caused by: java.lang.NullPointerException<br/>  at libcore.net.UriCodec.encode(UriCodec.java:132)<br/>  at java.net.URLEncoder.encode(URLEncoder.java:57)<br/>  at com.android.volley.Request.encodeParameters(Request.java:406)<br/>   at com.android.volley.Request.getBody(Request.java:392)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.addBodyIfExists(WPDelayedHurlStack.java:251)<br/>    at org.wordpress.android.networking.WPDelayedHurlStack.setConnectionParametersForRequest(WPDelayedHurlStack.java:238)<br/>  at org.wordpress.android.networking.WPDelayedHurlStack.performRequest(WPDelayedHurlStack.java:119)<br/> at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:93)<br/>    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:105)<br/>
45 - Stats Summary Call responded
46 - service destroyed
47 - No accounts configured.  Sending user to set up an account
48 - NewAccountAbstractOage.onCreate()
49 - GCM Unregistered ID: 
50 - instantiate XMLRPCClient using sFactory: class org.xmlrpc.android.XMLRPCFactoryDefault
51 - instantiate XMLRPCClient using sFactory: class org.xmlrpc.android.XMLRPCFactoryDefault
52 - instantiate XMLRPCClient using sFactory: class org.xmlrpc.android.XMLRPCFactoryDefault
53 - Cannot load blog options (wp.getOptions failed and no jetpack_client_id is then available - exception: null
54 - StackTrace: java.lang.NullPointerException<br/>    at org.wordpress.android.util.stats.AnalyticsTrackerMixpanelInstructionsForStat.addSuperPropertyToFlag(AnalyticsTrackerMixpanelInstructionsForStat.java:103)<br/>   at org.wordpress.android.util.stats.AnalyticsTrackerMixpanel.instructionsForStat(AnalyticsTrackerMixpanel.java:383)<br/>    at org.wordpress.android.util.stats.AnalyticsTrackerMixpanel.track(AnalyticsTrackerMixpanel.java:40)<br/>   at org.wordpress.android.util.stats.AnalyticsTrackerMixpanel.track(AnalyticsTrackerMixpanel.java:35)<br/>   at org.wordpress.android.util.stats.AnalyticsTracker.track(AnalyticsTracker.java:78)<br/>   at org.wordpress.android.ui.stats.StatsActivity$2.onSuccess(StatsActivity.java:232)<br/>    at org.xmlrpc.android.XMLRPCClient$Caller.run(XMLRPCClient.java:441)<br/>
55 - instantiate XMLRPCClient using sFactory: class org.xmlrpc.android.XMLRPCFactoryDefault
56 - instantiate XMLRPCClient using sFactory: class org.xmlrpc.android.XMLRPCFactoryDefault

--- a/src/org/wordpress/android/ui/stats/service/StatsService.java
+++ b/src/org/wordpress/android/ui/stats/service/StatsService.java
@@ -148,8 +148,9 @@ public class StatsService extends Service {
 
                 // visitors and views
                 String path = String.format("sites/%s/stats", blogId);
-                SummaryRestListener summaryListener = new SummaryRestListener(blogId);
+                SummaryListener summaryListener = new SummaryListener(blogId);
                 statsNetworkRequests.add(restClientUtils.get(path, summaryListener, summaryListener));
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
 
                 // bar charts
                 path = "batch/?urls%5B%5D=" 
@@ -161,6 +162,7 @@ public class StatsService extends Service {
                         getBarChartPath(blogId, StatsBarChartUnit.WEEK, 30), StatsBarChartUnit.WEEK,
                         getBarChartPath(blogId, StatsBarChartUnit.MONTH, 30), StatsBarChartUnit.MONTH);
                 statsNetworkRequests.add(restClientUtils.get(path, barChartRestListener, barChartRestListener));
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
 
                 // top posts and pages
                 path = "batch/?urls%5B%5D=" 
@@ -174,7 +176,8 @@ public class StatsService extends Service {
                         );
                 statsNetworkRequests.add(restClientUtils.get(path, topPostAndPageRestListener,
                         topPostAndPageRestListener));
-
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
+                
                 // referrers
                 path = "batch/?urls%5B%5D="
                         + Uri.encode(String.format("/sites/%s/stats/referrers?date=%s", blogId, today))
@@ -186,7 +189,8 @@ public class StatsService extends Service {
                         String.format("/sites/%s/stats/referrers?date=%s", blogId, yesterday)
                         );
                 statsNetworkRequests.add(restClientUtils.get(path, referrersListener, referrersListener));
-
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
+                
                 // clicks
                 path = "batch/?urls%5B%5D="
                         + Uri.encode(String.format("/sites/%s/stats/clicks?date=%s", blogId, today))
@@ -198,7 +202,8 @@ public class StatsService extends Service {
                         String.format("/sites/%s/stats/clicks?date=%s", blogId, yesterday)
                         );
                 statsNetworkRequests.add(restClientUtils.get(path, clicksListener, clicksListener));
-
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
+                
                 // search engine terms
                 path = "batch/?urls%5B%5D="
                         + Uri.encode(String.format("/sites/%s/stats/search-terms?date=%s", blogId, today))
@@ -211,7 +216,8 @@ public class StatsService extends Service {
                         );
                 statsNetworkRequests.add(restClientUtils
                         .get(path, searchEngineTermsListener, searchEngineTermsListener));
-
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
+                
                 // views by country - put at the end since this will start other networks calls on
                 // finish
                 path = "batch/?urls%5B%5D="
@@ -224,7 +230,8 @@ public class StatsService extends Service {
                         String.format("/sites/%s/stats/country-views?date=%s", blogId, yesterday)
                         );
                 statsNetworkRequests.add(restClientUtils.get(path, viewsByCountryListener, viewsByCountryListener));
-
+                AppLog.i(T.STATS, "Enqueuing the following Stats request " + path);
+                
                 numberOfNetworkCalls = statsNetworkRequests.size();
 
                 while (!isDone()) {
@@ -687,11 +694,11 @@ public class StatsService extends Service {
     }
 
     
-    private class SummaryRestListener implements RestRequest.Listener, RestRequest.ErrorListener {
+    private class SummaryListener implements RestRequest.Listener, RestRequest.ErrorListener {
         
         final String mRequestBlogId;
         
-        SummaryRestListener(String blogId) {
+        SummaryListener(String blogId) {
             this.mRequestBlogId = blogId;
         }
         

